models/migrations/v78.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-xorm/xorm"
+)
+
+func renameRepoIsBareToIsEmpty(x *xorm.Engine) error {
+	type Repository struct {
+		ID      int64 `xorm:"pk autoincr"`
+		IsBare  bool
+		IsEmpty bool `xorm:"INDEX"`
+	}
+
+	if err := x.Sync(new(Repository)); err != nil {
+		return fmt.Errorf("sync: %v", err)
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	_, err := sess.Query("UPDATE `repository` SET `is_empty` = `is_bare`")
+	if err != nil {
+		if strings.Contains(err.Error(), "no such column") {
+			return nil
+		}
+		return fmt.Errorf("select repositories: %v", err)
+	}
+	return sess.Commit()
```

Comment:
above this return can you put:
```
// Commit and begin new transaction for dropping columns
	if err := sess.Commit(); err != nil {
		return err
	}
	if err := sess.Begin(); err != nil {
		return err
	}

	if err := dropTableColumns(sess, "repository", "is_bare"); err != nil {
		return err
}
```
As then we can drop the `is_bare` column.

Reply:
Actually why don't we just drop the column in the same session?

Reply:
Sadly we can't do that because MSSQL errors out when you drop a column that is used/update in the same session.

Reply:
oh the stupidity - oh the unfathomable stupidity!

Reply:
😂

Reply:
Done